config/filter.d/mattermost.conf

```diff
@@ -0,0 +1,15 @@
+# Fail2Ban filter to match failed userlogin attempts as notified by mattermost's logfile
+#
+# Mattermost has to be configured to use timestamp at the start of the line. This is done in 'config/config.json'.
+# "LogSettings -> FileJson" needs to be set to 'false'
```

Comment:
Just curious, how it looks if `LogSettings -> FileJson` is not disabled?

Reply:
It looks like this:
```
{"level":"error","ts":1562574795.6335213,"caller":"mlog/log.go:172","msg":"Error trying to authenticate MFA token","path":"/api/v4/users/login","request_id":"xxx","ip_addr":"12.34.56.78","user_id":"xxx","method":"POST","err_where":"ValidateToken","http_code":401,"err_details":"invalid code"}

{"level":"error","ts":1561957767.752643,"caller":"mlog/log.go:172","msg":"Login failed because of invalid password","path":"/api/v4/users/login","request_id":"xxxx´","ip_addr":"12.34.56.78","user_id":"xxx","method":"POST","err_where":"checkUserPassword","http_code":401,"err_details":"user_id=xxx"}

{"level":"error","ts":1562571047.2427917,"caller":"mlog/log.go:172","msg":"Unable to find an existing account matching your credentials. This team may require an invite from the team owner to join.","path":"/api/v4/users/login","request_id":"xxx","ip_addr":"12.34.56.78","user_id":"","method":"POST","err_where":"GetUserForLogin","http_code":400,"err_details":""}
```

Which is why i opened https://github.com/fail2ban/fail2ban/issues/2463 in the first place :)

Reply:
And I answered you there how it may be possible...
So for this examples the parameters for fail2ban-regex look like:
```
msg='...'
fail2ban-regex -vv --VD -d '^\{"level"\s*:\s*"error",\s*"ts":{Epoch}\d*,' "$msg" '^\s*(?:"\w+"\s*:\s*(?:"[^"]*"|\S+)\s*,\s*)*"ip_addr":\s*"<ADDR>"\s*,\s*(?:"\w+"\s*:\s*(?:"[^"]*"|\S+)\s*,\s*)*"err_where"\s*:\s*"(?:checkUserPassword|GetUserForLogin|ValidateToken)"'
```
Note the `\d*` after `{Epoch}` - the float part is larger as 6 (fail2ban's epoch is microsecond precise).

Reply:
should add them both to the .conf then :)
will update the filter.d conf and add log line examples tomorrow